test/test_favorite_api.js

```diff
@@ -0,0 +1,59 @@
+const app = require('../lib/server').app;
+const express = require('express');
+const favoriteAPI = require('../lib/routes/favorite/favorite.controller');
+const User = require('../lib/models/user');
+const assert = require('assert');
+
+/*
+ NOTE: This test should work with the current Fall 2015 catalog that gets scraped.
+ It will not work with other data.
+ Additionally, this test code only works if the favorite table is already empty.
+
+*/
+/*
+request(app)
+.post('/api/favorites')
+.field('email', 'jabillings@middlebury.edu')
+.field('password', 'julian')
+.end(function(err, res) {
+  if (err) console.log(err);
+  console.log(res.cookie);
+});
+*/
+
+runTests = function(email){
+  return new Promise(function(resolve, reject) {
+    var user;
+    var usr = new User(email).fetch()
+    .then(function(usr) {
+      user = usr;
+      return favoriteAPI.getAllFavoritesHelper(user);
```

Comment:
Yeah, they're exported. I was having trouble with doing in-code authentication so I switched to making sure the underlying API code was functioning